### PR TITLE
test: create a test suite for examples

### PR DESCRIPTION
**Pull request by yoavz**
## Description
This PR creates a test suite associated with `examples`. It runs a minimal test on each example using `--local --test` mode. It is slightly more heavyweight than our unit tests since every example runs in a separate Python process, but more portable than full integration tests since no Determined cluster is required.

Run-time on my local macbook pro is 1 minute 42 seconds:

```
41.38s call     test_examples.py::test_official[official/cifar10_cnn_tf_keras-official/cifar10_cnn_tf_keras/const.yaml]
22.36s call     test_examples.py::test_official[official/cifar10_cnn_pytorch-official/cifar10_cnn_pytorch/const.yaml]
9.46s call     test_examples.py::test_official[official/multiple_lr_schedulers_pytorch-official/multiple_lr_schedulers_pytorch/const.yaml]
9.39s call     test_examples.py::test_official[official/mnist_pytorch-official/mnist_pytorch/const.yaml]
8.35s call     test_examples.py::test_official[official/mnist_estimator-official/mnist_estimator/const.yaml]
6.80s call     test_examples.py::test_official[official/fashion_mnist_tf_keras-official/fashion_mnist_tf_keras/const.yaml]
4.51s call     test_examples.py::test_official[official/iris_tf_keras-official/iris_tf_keras/const.yaml]
```

The runtime can likely further be improved by further optimizing `--local --test` mode with https://determinedai.atlassian.net/browse/DET-2931 

[DET-3110](https://determinedai.atlassian.net/browse/DET-3110)
[DET-2952](https://determinedai.atlassian.net/browse/DET-2952)

## Test Plan
N/A

## Commentary (optional)
I had initially aimed to set-up these as unit tests running sequentially in a single Python process with `test_one_batch`. However, I hit some problems with conflicting module namespaces in examples -- for example, `fashion_mnist_tf_keras` and `mnist_pytorch` have a `data.py` module they import in their entrypoint module with `import data`. When executed sequentially in a single Python process, the second time `import data` is called it still uses the initial `data` module and fails. I figured it would be easier to execute each model definition in a separate process instead of removing this restriction or hacking the Python module namespace in some way. 

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by brain-good** [comment]
@yoavz let's add this to CircleCI?

**Comment by yoavz** [comment]
@brain-good this is an initial pass at having "unit" tests for examples, with the caveat that I wasn't quite able to get them running all in the same Python process. 

Let me know if this approach looks good to you and I'll add the Circle CI target and remove the redundant tests in `e2e_tests`.

File: examples/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+@pytest.mark.parametrize("model_def,config_file", official_examples)
+def test_official(model_def: str, config_file: str) -> None:
+    examples_dir = pathlib.Path(__file__).parent
+    model_def_absolute = examples_dir.joinpath(model_def)
+    config_file_absolute = examples_dir.joinpath(config_file)
+
+    subprocess.check_output(
+        ("det", "experiment", "create", "--local", "--test", str(config_file_absolute), str(model_def_absolute)))
```
Comment (brain-good) [inline examples/test_examples.py:33]: question: do we lose the logs from running this?

File: examples/test_examples.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import os
+import pathlib
+import pytest
+import subprocess
+from ruamel import yaml
+
+from determined import experimental, load
```
Comment (brain-good) [inline examples/test_examples.py:7]: blocking: it seems that these are not used

we probably want to add flake8 in for linting the tests

**Comment by brain-good** [comment]
Sounds good @yoavz , think the approach makes sense 👍

File: .circleci/config.yml [hunk 3]
```diff
@@ -767,6 +767,18 @@ jobs:
...
       - run: make -C harness test-tf2
 
+  test-examples:
+    docker:
+      - image: determinedai/cimg-base:stable
+    steps:
+      - checkout
+      - setup-python-venv:
+          determined-common: true
+          determined: true
```
Comment (brain-good) [inline .circleci/config.yml:777]: you're probably going to need the CLI

File: examples/test_examples.py (hunk 1 above)
Comment (yoavz) [reply examples/test_examples.py]: No, pytest will output logs and a stacktrace if there is a failure.

**Pull request merged**